qiime2/sdk/usage.py

```diff
@@ -922,6 +922,146 @@ def init_format(self, name: str,
         """
         return self._usage_variable(name, factory, 'format')
 
+    def _replace_url_epoch(self, url):
+        # Obtaining epoch modeled on qiime2.metadata.io.MetadataFileError
+        import qiime2
+
+        epoch = qiime2.__release__
+        return url.replace('{epoch}', epoch)
+
+    def _request_url(self, url):
+        import requests
```

Comment:
It's interesting - the `requests` module isn't being found when CI is running your tests. I'm wondering if it actually does need to be added at the top of the file, or maybe in the recipe file as well.

Reply:
Oh, I wonder if it's not actually a framework dependency?

Reply:
I might recommend using stdlib for this (even if it's slightly unpleasant), since there's not too much going on. If we were doing lots of URL requests, then a 3rd party lib might be a good choice.

Since almost everyone uses requests, it's easy for us to end up in a dependency bind someday, so not using it would avoid that.